File: URI/1 - INICIANTE/C/1164 - NumeroPerfeito.c
#include <stdio.h>

int main(){
	int x = 0, num, cont, soma, z;
	scanf(" %d", &z);
	while(z > 0){
		scanf(" %d", &num);
		soma = 0;
		for(cont = 1; cont < num; cont++){
			if(num % cont == 0){
				soma += cont;
			}
		}
		if(soma == num){
			printf("%d eh perfeito\n", num);
		}else{
			printf("%d nao eh perfeito\n", num);
		}
		z --;		
	}
	
	return 0;
}

File: URI/1 - INICIANTE/Python/1164 - NumeroPerfeito.py
z = int(input())
while z > 0:
    num = int(input())
    soma = 0
    for y in range(1, num):
        if num % y == 0:
            soma += y
    if soma == num:
        print('{} eh perfeito'.format(num))
    else:
        print('{} nao eh perfeito'.format(num))
    z -= 1